src/pip/_internal/utils/direct_url_helpers.py

```diff
@@ -77,3 +86,24 @@ def direct_url_from_link(link, source_dir=None, link_is_in_wheel_cache=False):
             direct_url.info.hash = "{}={}".format(hash_name, link.hash)
     direct_url.subdirectory = link.subdirectory_fragment
     return direct_url
+
+
+def dist_get_direct_url(dist):
+    # type: (Distribution) -> Optional[DirectUrl]
+    """Obtain a DirectUrl from a pkg_resource.Distribution.
+
+    Returns None if the distribution has no `direct_url.json` metadata,
+    or if `direct_url.json` is invalid.
+    """
+    if not dist.has_metadata(DIRECT_URL_METADATA_NAME):
+        return None
+    try:
+        return DirectUrl.from_json(dist.get_metadata(DIRECT_URL_METADATA_NAME))
+    except Exception as e:
```

Comment:
I initially wanted to catch `DirectUrlvalidationError, JSONDecodeError, UnicodeDecodeError` which are those that signal an invalid `direct_url.json`. Unfortunately `JSONDecodeError` is not in python 2. I'd leave out `IOError` because that's probably an error we'd want to be fatal.

Any suggestion for catching JSON decoding errors in python 2?

Reply:
On Python 2 the best you can do is to catch `ValueError` (which is also a documented superclass to `json.JSONDecodeError` on Python 3). I would make the try block only contain the `load` line and raise a custom `DirectURLRecordError` to make sure we don’t accidentally catch unrelated `ValueError`.

Reply:
@uranusjr thanks for the hint about `ValueError`. I ended up making an alias for JSDONDecodeError on PY2, so the code remains clean and will not need to change when deprecating python 2 in the future.